de.metas.marketing/base/src/main/java/de/metas/marketing/base/model/interceptor/AD_User.java

```diff
@@ -42,26 +43,33 @@
  * #L%
  */
 @Interceptor(I_AD_User.class)
-@Component
+@Component("de.metas.marketing.base.model.interceptor.AD_User")
 public class AD_User
 {
+
 	private final static String MRG_MKTG_Campaign_NewsletterGroup_Missing_For_Org = "MKTG_Campaign_NewsletterGroup_Missing_For_Org";
 
+	private final CampaignRepository campaignRepository;
+	private final CampaignService converters;
 	private final UserRepository userRepository;
+	private final ContactPersonService contactPersonService;
 
-	private AD_User(@NonNull final UserRepository userRepository)
+	private AD_User(@NonNull final ContactPersonService contactPersonService,
```

Comment:
I didn't even know this worked ( https://stackoverflow.com/questions/7254496/java-spring-bean-with-private-constructor ), but why did you make it private?